app/lib/bk/compat/parsers/bitrise/github_release.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Handles the translation of github-release
+      class Translator
+        def translate_github_release(inputs)
+          raise 'Invalid github-release step configuration!' unless validate_required_params(inputs)
+
+          release_config = extract_release_config(inputs)
+          owner, repo_name = parse_repo(release_config[:repository_url])
+
+          commands = build_commands(release_config, owner, repo_name)
+          BK::Compat::CommandStep.new(commands: commands)
+        end
+
+        private
+
+        def validate_required_params(inputs)
+          %w[api_token username repository_url tag commit name].all? { |key| inputs.include?(key) }
+        end
+
+        def extract_release_config(inputs)
+          {
+            api_token: inputs['api_token'],
+            commit: inputs['commit'],
+            name: inputs['name'],
+            repository_url: inputs['repository_url'],
+            tag: inputs['tag'],
+            body: inputs['body'] || '',
+            draft: inputs['draft'] || 'yes',
+            pre_release: inputs['pre_release'] || 'no',
+            files_to_upload: inputs['files_to_upload'],
+            api_base_url: inputs['api_base_url'] || 'https://api.github.com',
+            upload_base_url: inputs['upload_base_url'] || 'https://uploads.github.com',
+            generate_release_notes: inputs['generate_release_notes'] || 'no'
+          }
+        end
+
+        def parse_repo(url)
+          url.chomp!('.git')
+
+          case url
+          when %r{^https://}, /^git@/, %r{^ssh://}
+            url = url.sub(%r{^(https://|git@|ssh://)}, '')
+            _, repo = url.split(%r{[:/]}, 2)
+          end
+
+          owner, repo_name = repo&.split('/') || []
+          raise 'Invalid repository URL!' if owner.nil? || repo_name.nil?
+
+          [owner, repo_name]
+        end
```

Comment:
this extraction is only done to later be re-joined so I don't think it needs to be so thorough. Usin ruby's builtin `uri` library would also make it a lot more robust and simpler:

```suggestion
        def parse_repo(url)
          data = URI.parse(url.chomp('.git'))
          
          data.path
        end
```